server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -793,6 +794,14 @@ protected Node(
                 remoteStoreStatsTrackerFactory
             );
 
+            final PerfStatsCollectorService perfStatsCollectorService = new PerfStatsCollectorService(clusterService);
+            final NodePerformanceTracker nodePerformanceTracker = new NodePerformanceTracker(
```

Comment:
I didn't want to put too many responsibilities in 'PerfStatsCollectorService' - because later it will be used for coordinator AC rejections + ranking as well. In this way its a bit cleaner i feel.